Dockerfile

```diff
@@ -12,10 +12,21 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-FROM golang:1.23-bullseye as builder
+FROM --platform=$BUILDPLATFORM golang:1.23-bullseye AS builder
+
+ARG TARGETOS
+ARG TARGETARCH
+
 WORKDIR /go/src/github.com/GoogleCloudPlatform/container-engine-accelerators
 COPY . .
-RUN go build cmd/nvidia_gpu/nvidia_gpu.go
+RUN if [ "${TARGETARCH}" = "arm64" ] && [ "${BUILDARCH}" != "arm64" ]; then \
```

Comment:
can we double check it conditions here? Is it assuming the build machine is on X86?

Reply:
Yes the current build machine is x86. These conditions are ensuring that we only install the cross-compiler if the host machine is x86. It may be a good point to ensure that the host machine is nothing other than x86 or arm though, if that is what you were getting at?